File: 0301_0400/0392/0392.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    f.readline()
    s = f.readline().strip()
    i = s.find(' 1 ')
    q.write(s[i + 1:] + ' ' + s[:i] if i > -1 else s if s[:2] == '1 ' else '1 ' + s[:-1])

File: 0301_0400/0392/0392.c
#include <stdio.h>
#include <string.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n;
	char s[600000], *m;

	fscanf(f, "%d\n", &n);
	fgets(s, 599999, f);
	n = strlen(s) - 1;
	while (s[n] == ' ' || s[n] == '\n')
		--n;
	s[n + 1] = '\0';
	m = strstr(s, " 1 ");
	if (m)
	{
		*m = '\0';
		fprintf(q, "%s ", m + 1);
	}
	else if (s[0] != '1' || s[1] != ' ')
	{
		s[n] = '\0';
		fprintf(q, "1 ");
	}
	fprintf(q, "%s", s);
	return 0;
}
